Simplified/Reader2/UI/NYPLBaseReaderViewController.swift

```diff
@@ -261,7 +263,40 @@ class NYPLBaseReaderViewController: UIViewController, Loggable {
       updateBookmarkButton(withState: false)
     }
   }
+  
+  //----------------------------------------------------------------------------
+  // MARK: - Read Position
 
+  private func restoreReadPosition() {
+    bookmarksBusinessLogic.restoreReadPosition(currentLocator: navigator.currentLocation, localFetchCompletion: { (localLocator) in
+      if let localLocator = localLocator {
+        self.navigator.go(to: localLocator, animated: true) {}
+      }
+    }) { [weak self] (serverLocator) in
+      if let serverLocator = serverLocator {
+        self?.presentReadPositionNavigationAlert(locator: serverLocator)
```

Comment:
make sure to call this from main thread (you can use NYPLMainThreadRun helper) -- same thing for the `navigator.go(to:)` call